src/components/input/RadioGroup.tsx

```diff
@@ -0,0 +1,120 @@
+import classnames from 'classnames';
+import { useRef } from 'preact/hooks';
+
+import { useArrowKeyNavigation } from '../../hooks/use-arrow-key-navigation';
+import { RadioCheckedIcon, RadioIcon } from '../icons';
+
+export type RadioInput<T extends string | number> = {
+  value: T;
+  label: string;
+  subtitle?: string;
+  disabled?: boolean;
+};
+
+export type RadioGroup<T extends string | number> = {
+  inputs: RadioInput<T>[];
+  selected?: T;
+  onChange: (newSelected: T) => void;
+
+  /** Determines the direction in which radios are stacked */
+  direction?: 'vertical' | 'horizontal';
+
+  disabled?: boolean;
+  'aria-label'?: string;
+  'aria-labelledby'?: string;
+
+  /**
+   * If provided, it adds an input[type="hidden"] to the DOM with selected value,
+   * allowing to use this in forms as a regular form control
+   */
+  name?: string;
+};
+
+function Radio<T extends string | number>({
+  input: { value, label, subtitle, disabled },
+  isSelected,
+  onChange,
+}: {
+  input: RadioInput<T>;
+  isSelected: boolean;
+  onChange: (newSelected: T) => void;
+}) {
+  return (
+    <div
```

Comment:
It looks to me like all the necessary semantic attributes have been added. You decided not to go with a hidden input?

Reply:
I'm not sure I follow the question.

Reply:
Please ignore this. I was asking why you chose not to use a hidden radio input like the `RadioButton` component but I think that's answered in the PR description.